src/pages/components/Demo/Demo.js

```diff
@@ -23,6 +23,9 @@ const Demo = ({ searchWord, words }) => {
         window.location.hash = 'try-it-out';
       }
     }
+    setRedirectUrl(window.location.hostname === 'localhost'
```

Comment:
This is might throw an error on the homepage if `window` is undefined (take a look at line 18). `window` might be undefined since this component is rendered on the server side where the `window` object doesn't exist.

If you want to reference `window`, meaning you're in the client, then you want to do so in the `if (typeof window !== 'undefined')` block. But we want to determine the correct link when in the server, where we don't have access to `window`. We would need to do that using environment variables.

So the ideal solution would be, determine the correct link for both the server side and the client side. Take a look at the [NextJS Environment Variable docs](https://nextjs.org/docs/basic-features/environment-variables) for some help